### PR TITLE
ceph.spec.in: Fixes for el9 builds

### PR DESCRIPTION
**Pull request by djgalloway**
## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by djgalloway** [comment]
Building here: https://shaman.ceph.com/builds/ceph/el9/9b743e2cdcdc70af5938c9f0d16d5bdb9beee4b4/default/313672/

**Comment by tchaikov** [comment]
i created #47538. but i am fine either way.

**Review by tchaikov — APPROVED** [review]

File: ceph.spec.in [hunk 1]
```diff
@@ -153,7 +153,7 @@
 # do not provide gcc-annobin.so anymore, despite that they provide annobin.so. but
 # redhat-rpm-config still passes -fplugin=gcc-annobin to the compiler.
 %undefine _annotated_build
-%if 0%{?rhel} >= 8
+%if 0%{?rhel} == 8
```
Comment (tchaikov) [inline ceph.spec.in:156]: i believe that https://github.com/ceph/ceph/pull/47538/commits/75dd92353381c6ec5ac7251585a276059b8a2253 is a better fix though.

Comment (djgalloway) [reply ceph.spec.in:156]: Will it install gcc-toolset-12 on el9 though?

Comment (tchaikov) [reply ceph.spec.in:156]: please see the commit message of https://github.com/ceph/ceph/pull/47538/commits/edf07165a44eb73f3bbbae0c300d7f4f1bba1204 in  #47538

Comment (djgalloway) [reply ceph.spec.in:156]: gcc-toolset-11 is not available in CentOS 9 though.  Does that not matter?

Comment (djgalloway) [reply ceph.spec.in:156]: > gcc-toolset-11 is not available in CentOS 9 though. Does that not matter?

Guess not.  It built.

**Comment by djgalloway** [comment]
Closed in favor of https://github.com/ceph/ceph/pull/47538

**Pull request closed without merging**